src/view/mod.rs

```diff
@@ -10,39 +14,220 @@ use tui::{
     Terminal,
 };
 
+#[derive(Debug)]
+pub struct StyleFunc {
+    pub f: fn(Style) -> Style,
+}
+
 pub struct View<B: Backend> {
     terminal: Terminal<B>,
     // Probably want to include layout details
     //  here eventually.
+    tag_to_func: HashMap<Tag, StyleFunc>,
 }
 
 impl<B: Backend> View<B> {
-    pub fn new(terminal: Terminal<B>) -> View<B> {
-        View { terminal: terminal }
+    pub fn new(terminal: Terminal<B>, tag_to_func: HashMap<Tag, StyleFunc>) -> View<B> {
+        View {
+            terminal: terminal,
+            tag_to_func: tag_to_func,
+        }
     }
 
     pub fn update_display(&mut self, app: &App) -> Result<(), io::Error> {
-        // Get a copy of the text to be rendered
-        // For now let's not do anything fancy formatting
-        let text = app.get_text_based_on_mode();
-
-        let text: Vec<_> = text.iter().map(|x| Text::raw(x)).collect();
-
-        let title = match app.app_mode() {
-            AppMode::Command(CommandMode::Write) => "Command mode: Saving into file",
-            AppMode::Command(CommandMode::Read) => "Command mode: Opening a file",
-            AppMode::Edit => "Edit mode",
-        };
-
+        let tagged_text = app.get_tagged_text(); // Get a copy of the text to be rendered
+                                                 // For now let's not do anything fancy formatting
+        let text: Vec<_> = highlighter::highlight_tagged_text(&tagged_text, &self.tag_to_func);
         self.terminal.draw(|mut f| {
             let size = f.size();
             let block = Paragraph::new(text.iter())
-                .block(Block::default().title(title).borders(Borders::ALL))
```

Comment:
Due to these changes, there is no visual indication for Open and Save file features anymore. The features still exist and it will work, but user cannot see that they are in the middle of saving/opening a file.

I think it should be a quick change for the `get_text_based_on_mode()` function to call `as_tagged_text()` on the correct buffer.

Reply:
oh i see. I'll add that back in. 

though I think the View component should be the one making the decision as to which buffer it is reading from, rather than having the app struct decide it. reason being the app have 0 regards as to what should be given/shown to View.

Reply:
Ah okay. My initial thoughts were quite the opposite; I wanted to make the View as "dumb" as possible so that it will only show what is given rather than making any decisions.

But I'm not fully sure what's a good practice. I'm all right with following what you have!